tests/unit/commands/test_list_registered.py

```diff
@@ -15,30 +15,41 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 from textwrap import dedent
-from unittest import mock
 
 from testtools.matchers import Contains, Equals
 
 from snapcraft import storeapi
-from . import CommandBaseTestCase
+from . import FakeStoreCommandsBaseTestCase
 
 
-class ListRegisteredTestCase(CommandBaseTestCase):
+class ListRegisteredTestCase(FakeStoreCommandsBaseTestCase):
 
     scenarios = [
         ("list-registered", {"command_name": "list-registered"}),
         ("registered alias", {"command_name": "registered"}),
     ]
 
     def test_command_without_login_must_ask(self):
-        result = self.run_command([self.command_name])
+        # TODO: look into why this many calls are done inside snapcraft.storeapi
```

Comment:
is this intended to be fixed in this PR? is this a regression in behavior or consistent because it wasn't checked before?

Reply:
Not related to the PR, I bet it is "refresh_if_necessary" in the storeapi. We never checked this before, we never seem to have checked many things before which now are properly checked.